sched/sched/sched_timerexpiration.c

```diff
@@ -238,7 +238,26 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
     {
       /* Apply the keep alive hack */
 
+#ifdef CONFIG_SCHED_RR_IDLETASK
       return KEEP_ALIVE_TICKS;
+#else
+
+      /* The idle task is the only task in that sched_priority == 0.
+       */
+
+      if (ntcb->sched_priority != 0)
+      {
+          return KEEP_ALIVE_TICKS;
+      }
+      else
+      {
+#ifdef CONFIG_SCHED_TICKLESS_LIMIT_MAX_SLEEP
+          return g_oneshot_maxticks - 1;
+#else
+          return UINT32_MAX - 1;
+#endif
```

Comment:
can return zero directly, since the caller will treat zero as uINT32_MAX.

Reply:
I found up_alarm_start will not call if we return 0 or UINT32_MAX directly. Someone maybe do some statistic job on this call?